aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1381,60 +1577,73 @@ Tensor _csr_to_block_csr_cpu(const Tensor& self, IntArrayRef blocksize) {
-  Tensor cpu_result = _csr_to_block_csr_cpu(
-      _sparse_csr_tensor_unsafe(
-          self_crow_indices.cpu(),
-          self_col_indices.cpu(),
-          self_values_cpu,
-          self.sizes(),
-          self_values.scalar_type(),
-          self.layout(),
-          self_values_cpu.device()),
-      blocksize);
-  Tensor result_values = cpu_result.values().to(self_values.options());
-  Tensor result_crow_indices =
-      cpu_result.crow_indices().to(self_crow_indices.options());
-  Tensor result_col_indices =
-      cpu_result.col_indices().to(self_col_indices.options());
-  return at::native::_sparse_bsr_tensor_unsafe(
-      result_crow_indices,
-      result_col_indices,
-      result_values,
-      self.sizes(),
-      result_values.scalar_type(),
-      c10::kSparseBsr,
-      result_values.device());
+  if (self.layout() == kSparseBsc) {
+    return sparse_compressed_to_flipped(self, blocksize, "to_sparse_bsr");
+  }
+  if (self.layout() == kSparseBsr) {
+    return sparse_compressed_clone(self, blocksize, "to_sparse_bsr");
+  }
+  if (self.layout() == kSparseCsr) {
+    TORCH_CHECK(self.dim() == 2,
+        "to_sparse_bsr(): conversion from Csr to Bsr is only possible for 2d inputs, ",
+        "but got input of dimension ", self.dim(), " instead.");
+    Tensor self_values = self.values();
+    Tensor self_crow_indices = self.crow_indices();
+    Tensor self_col_indices = self.col_indices();
+    Tensor cpu_result = _csr_to_block_csr_cpu(
+        _sparse_csr_tensor_unsafe(
+            self_crow_indices.cpu(),
+            self_col_indices.cpu(),
+            self_values.cpu(),
+            self.sizes(),
+            self_values.scalar_type(),
+            self.layout(),
+            at::kCPU),
+        blocksize);
+    Tensor result_values = cpu_result.values().to(self_values.options());
+    Tensor result_crow_indices =
+        cpu_result.crow_indices().to(self_crow_indices.options());
+    Tensor result_col_indices =
+        cpu_result.col_indices().to(self_col_indices.options());
+    return at::native::_sparse_bsr_tensor_unsafe(
+        result_crow_indices,
+        result_col_indices,
+        result_values,
+        self.sizes(),
+        result_values.scalar_type(),
+        c10::kSparseBsr,
+        result_values.device());
+  }
+  AT_ERROR(
+      "sparse_compressed_to_sparse_bsr expected SparseCsr, SparseBsr or SparseBsc layout but got ",
+      self.layout());
+  return self;
 }
 
 Tensor sparse_compressed_to_sparse_bsc(const Tensor& self, IntArrayRef blocksize) {
+  if (self.layout() == kSparseBsr) {
+    return sparse_compressed_to_flipped(self, blocksize, "to_sparse_bsr");
+  }
+  if (self.layout() == kSparseBsc) {
+    return sparse_compressed_clone(self, blocksize, "to_sparse_bsr");
+  }
   AT_ERROR(
-      "Conversion from ", self.layout(), " to SparseBsc is currently not supported.");
+      "sparse_compressed_to_sparse_bsc expected SparseBsr or SparseBsc layout but got ",
```

Comment:
We could add support for CSC->BSC using a transpose + conversion of CSR->BSR + transpose back to BSC. It would be 2d only though (since CSR->BSR is 2d only).

`self.transpose(-2, -1).to_sparse_bsr(blocksize).transpose(-2, -1)`

I am fine leaving it out though as it is not particularly useful, this comment is mainly to jog our memory if a request comes in for the functionality.